src/helpers.c

```diff
@@ -579,7 +612,44 @@ enum swupd_code swupd_init(enum swupd_init_config config)
 	}
 
 	if ((config & SWUPD_NO_NETWORK) == 0) {
-		if (swupd_curl_init() != 0) {
+
+		bool curl_init = true;
+
+		/* try to init curl with whatever server is set up */
+		if (swupd_curl_init(NULL) != 0) {
+			/* the server is not responding */
+			if (mirror_is_set()) {
+				/* the server is a mirror, check to see if the upstream server is working */
+				swupd_curl_deinit();
```

Comment:
if curl_init fails it shouldn't be in a state where it's inited. this is a bug in curl_init. we need to run denit in culr_init if there's any errors.

Reply:
I'll change that